lib/net.js

```diff
@@ -425,12 +421,14 @@ Socket.prototype._destroy = function(exception, cb) {
   var self = this;
 
   function fireErrorCallbacks() {
+    var hadSeenError = self._errorEmitted;
+    self._errorEmitted = self._errorEmitted || Boolean(exception);
```

Comment:
Ah, if you mean `hadSeenError`, we want to set that *before* any callbacks fire, to ensure that the outside world sees the stream in the proper state. However, we still want to inspect the original value and switch our behavior based on that.